### PR TITLE
Feature/button reaction change

### PR DESCRIPTION
**Pull request by thedarkrai07**

### CODE REVIEW

File: src/app/app.ts [hunk 1]
```diff
@@ -56,6 +56,7 @@ const creator = new SlashCreator({
 	applicationID: process.env.DISCORD_BOT_APPLICATION_ID,
 	publicKey: process.env.DISCORD_BOT_PUBLIC_KEY,
 	token: process.env.DISCORD_BOT_TOKEN,
+	disableTimeouts: true,
```
Comment (tescher) [inline src/app/app.ts:59]: What is this doing, exactly? Any other side effects?

File: src/app/events/InteractionCreate.ts [hunk 2]
```diff
@@ -21,42 +21,28 @@ import DMPermissionError from '../errors/DMPermissionError';
...
-                await user.fetch();
-            } catch (error) {
-                LogUtils.logError('failed to pull user partial', error);
-                return;
-            }
-        }
-
+    async execute(interaction: ButtonInteraction): Promise<any> {
+        const user: User = interaction.user;
+        if (!interaction.isButton()) return;
```
Comment (tescher) [inline src/app/events/InteractionCreate.ts:29]: What about existing bounty card messages that have reactions on them? We either need to convert them or still support emoji reactions.

I suggest supporting the reactions, but converting the cards to buttons when a user does a reaction. Will have a bunch of deprecated code, but not sure how else to do it.

File: src/app/activity/bounty/ActivityHandler.ts [hunk 3]
```diff
@@ -48,6 +48,16 @@ export const BountyActivityHandler = {
...
         Log.debug(request.activity)
+
+        if (request.commandContext) {
+            // If bot is not able to reply within 2 secs, defer reply
+            setTimeout(() =>{
+                if (!request.commandContext.initiallyResponded) {
+                    request.commandContext.defer();
+                }
+            }, 2000)
+        }
```
Comment (tescher) [inline src/app/activity/bounty/ActivityHandler.ts:59]: Good

**Review by tescher — CHANGES_REQUESTED** [review]
See comment - need to support existing messages that don't have buttons yet.

File: src/app/app.ts (hunk 1 above)
Comment (thedarkrai07) [reply src/app/app.ts:59]: Timeout differs the reply if it takes more than 2 seconds. I have added it back exactly in src/app/activity/bounty/ActivityHandler.ts. There are no side-effects for this

Comment (tescher) [reply src/app/app.ts:59]: OK, so by recreating the functionality you bypassed the discord error you were getting?

Comment (thedarkrai07) [reply src/app/app.ts:59]: Right

Comment (thedarkrai07) [reply src/app/app.ts:59]: Basically, we disabled that functionality from slash-create and added it back from our end

File: src/app/requests/RefreshRequest.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    commandContext: CommandContext;
+    message: Message;
+    buttonInteraction: ButtonInteraction;
+
+    constructor(args: {
+        commandContext: CommandContext,
+        messageReactionRequest: MessageReactionRequest,
+        buttonInteraction: ButtonInteraction,
+    }) {
+        if (args.commandContext) {
```
Comment (tescher) [inline src/app/requests/RefreshRequest.ts:20]: This code is unreachable I think. Should never be a command context for refresh

File: src/app/utils/DiscordUtils.ts [hunk 5]
```diff
@@ -185,6 +185,7 @@ const DiscordUtils = {
     },
 
     getBountyIdFromEmbedMessage(message: Message): string {
+        if (message.embeds[0].fields[BountyEmbedFields.bountyId].name !== 'Bounty Id') return null;
```
Comment (tescher) [inline src/app/utils/DiscordUtils.ts:188]: Was there an issue here you are trying to catch?

File: src/app/requests/RefreshRequest.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            if (args.commandContext.subcommands[0] !== Activities.apply) {
+                throw new Error('ApplyRequest attempted created for non Apply activity.');
```
Comment (tescher) [inline src/app/requests/RefreshRequest.ts:22]: Refresh, not Apply

**Review by tescher — CHANGES_REQUESTED** [review]
Couple more fixes needed. See comments

File: src/app/events/InteractionCreate.ts [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+        }
+
+        try {
+            await handler(request);
+        }
+        catch (e) {
+            let errorContent = e.message;
+            if (e instanceof NotificationPermissionError) {
+                return ErrorUtils.sendToDefaultChannel(e.message, request);
+            } else if (e instanceof ValidationError) {
```
[the remark this answers is not in the document]
Comment (thedarkrai07) [reply src/app/events/InteractionCreate.ts:218]: No, check line 234, 235

File: src/app/requests/RefreshRequest.ts (hunk 4 above)
Comment (thedarkrai07) [reply src/app/requests/RefreshRequest.ts:20]: Right. Can remove it

File: src/app/events/InteractionCreate.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (tescher) [reply src/app/events/InteractionCreate.ts:218]: Huh, I could swear it was failing with no messages for me. Now it working. OK, good

**Review by tescher — COMMENTED** [review]
Couldn't reproduce the silent error :)

File: src/app/requests/RefreshRequest.ts (hunk 6 above)
Comment (thedarkrai07) [reply src/app/requests/RefreshRequest.ts]: Fixed

File: src/app/requests/RefreshRequest.ts (hunk 4 above)
Comment (thedarkrai07) [reply src/app/requests/RefreshRequest.ts]: fixed

**Review by tescher — APPROVED** [review]

**Pull request merged**